pkg/daemon/ceph/osd/volume.go

```diff
@@ -582,9 +582,15 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		}
 		mdArgs = append(mdArgs, strings.Split(conf["devices"], " ")...)
 
+		// Do not change devices passed by-id
+		mdPath := path.Join("/dev", md)
+		if strings.Contains(md, "by-") {
+			mdPath = md
```

Comment:
strings.Prefix("/dev/") is simpler. So how about the following code?

```
if mdPath := md
if !strings.Prefix("/dev") {
  mdpath := path.Join("/dev", md)
}
```

Reply:
More generic too! Thanks!